src/libNode/Node.cpp

```diff
@@ -2352,13 +2352,21 @@ void Node::CleanCreatedTransaction() {
     m_createdTxns.clear();
     t_createdTxns.clear();
   }
-  {
-    std::lock_guard<mutex> g(m_mutexTxnPacketBuffer);
-    m_txnPacketBuffer.clear();
-  }
-  {
-    std::lock_guard<mutex> g(m_mutexTxnPktInProcess);
-    m_txnPktInProcess.clear();
+  // Avoid cleaning when buffer already have packets waiting to be picked up for
+  // distribution. Thus avoid deadlock.
+  if (m_txnPacketThreadOnHold == 0) {
+    {
+      // extra safety
+      std::unique_lock<std::mutex> lock(m_mutexTxnPacketBuffer,
+                                        std::try_to_lock);
+      if (!lock.owns_lock()) {
+        m_txnPacketBuffer.clear();
```

Comment:
Maybe before every `clear` in this function we can add a short log message so we know if and when the different clearings are done.